src/test/resources/schema/issue526.schema.json

```diff
@@ -0,0 +1,64 @@
+{
+  "$id": "https://schemas.example.com/example-service/commands/1.0.0/example.schema.json",
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "title": "example",
+  "type": "object",
+  "properties": {
+    "phones": {
+      "description": "The example phone numbers",
+      "type": "array",
+      "items": {
+        "$ref": "https://schemas.example.com/example-service/objects/1.0.0/phone.schema.json",
```

Comment:
For refs that are inside the same file, we use it like this:
`"$ref": "#/definitions/Policy"`

with the definition looking like:

```
"definitions": {
    "Policy": {
      "type": "object",
      "properties": {
        "id": {
      ...
```

The `#` is used for stuff relative to the same file.

The full file is [here](https://github.com/RedHatInsights/event-schemas/blob/main/schemas/apps/policies/v1/policy-triggered.json#L12).

If you want to use ids that are not real urls, you could implement an URI factory like:

```java
static class CloudEventURIFactory implements URIFactory {

        private static final String baseUrl = "https://console.redhat.com/api";
        private final URLFactory urlFactory = new URLFactory();
        private final String base;

        CloudEventURIFactory() {
            String fullPath = RHELSystem.class.getResource(schemaPath).toString();
            base = fullPath.substring(0, fullPath.length() - schemaPath.length());
        }

        @Override
        public URI create(String uri) {
            return urlFactory.create(replaceBase(uri));
        }

        @Override
        public URI create(URI baseURI, String segment) {
            return urlFactory.create(URI.create(replaceBase(baseURI.toString())), segment);
        }

        private String replaceBase(String uri) {
            if (uri.startsWith(baseUrl)) {
                uri = base + uri.substring(baseUrl.length(), uri.length() - 1);
            }

            return uri;
        }
    }
```

and adding it to your JsonSchemaFactory: 

```
JsonSchemaFactory.Builder().
                .uriFactory(new CloudEventURIFactory(), "https")
                .build();
```

Full example here: https://github.com/RedHatInsights/event-schemas-java/blob/main/src/main/java/com/redhat/cloud/event/parser/ConsoleCloudEventParser.java#L35C15-L63

I hope that above information is useful, I didn't go a lot into the details.